driver-core/src/main/com/mongodb/internal/connection/InternalStreamConnectionFactory.java

```diff
@@ -48,20 +50,22 @@ class InternalStreamConnectionFactory implements InternalConnectionFactory {
             @Nullable final MongoCredentialWithCache credential,
             @Nullable final String applicationName, final MongoDriverInformation mongoDriverInformation,
             final List<MongoCompressor> compressorList,
-            @Nullable final CommandListener commandListener, @Nullable final ServerApi serverApi) {
+            final LoggerSettings loggerSettings, @Nullable final CommandListener commandListener, @Nullable final ServerApi serverApi) {
         this(clusterConnectionMode, false, streamFactory, credential, applicationName, mongoDriverInformation, compressorList,
-                commandListener, serverApi);
+                loggerSettings, commandListener, serverApi);
     }
+
     InternalStreamConnectionFactory(final ClusterConnectionMode clusterConnectionMode, final boolean isMonitoringConnection,
-                                    final StreamFactory streamFactory,
-                                    @Nullable final MongoCredentialWithCache credential,
-                                    @Nullable final String applicationName, final MongoDriverInformation mongoDriverInformation,
-                                    final List<MongoCompressor> compressorList,
-                                    @Nullable final CommandListener commandListener, @Nullable final ServerApi serverApi) {
+            final StreamFactory streamFactory,
+            @Nullable final MongoCredentialWithCache credential,
+            @Nullable final String applicationName, final MongoDriverInformation mongoDriverInformation,
+            final List<MongoCompressor> compressorList,
+            final LoggerSettings loggerSettings, @Nullable final CommandListener commandListener, @Nullable final ServerApi serverApi) {
         this.clusterConnectionMode = clusterConnectionMode;
```

Comment:
[optional] When a method header must be broken due to length (exceeds 120), could we as a rule place each parameter (including the first) on a new line, and order the assignments in parameter order? (For the same reasons that we do not put all fields and assignments on the same line.)

This also makes it easier to scan and correlate: for example, there is a null check on compressorList, but loggerSettings is not marked `@Nullable` and has no such check (and the same is the case for clusterConnectionMode).